nav2_behaviors/include/nav2_behaviors/plugins/drive_on_heading.hpp

```diff
@@ -168,6 +169,10 @@ class DriveOnHeading : public TimedBehavior<ActionT>
     const geometry_msgs::msg::Twist & cmd_vel,
     geometry_msgs::msg::Pose2D & pose2d)
   {
+    if (command_no_collision_checks_) {
```

Comment:
You should use the action goal's value here, not create a new class member variable :-)

Reply:
I tried to follow same pattern used for `command_x_` and `command_speed_`. Otherwise, I will have to pass the action goal to this method. Which do you prefer? Thanks.

Reply:
But you didn't populate it, it seems?